src/app/store/reference-data/effects/reference-data.effects.ts

```diff
@@ -52,6 +58,35 @@ export class ReferenceDataEffects {
     )
   );
 
+  fetchReferenceDataByKeySearch$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(fetchReferenceDataByKeySearch),
+      mergeMap(({ resourceType, resourceKey }) =>
+        this.referenceDataService.fetchReferenceDataByKeySearch(resourceType, resourceKey).pipe(
+          map(data => fetchReferenceDataByKeySearchSuccess({ resourceType, resourceKey, payload: data.data as ReferenceDataModelBase[] })),
```

Comment:
if you pull latest code, there's a new observable operator you can use to handle event where data was not found.

Reply:
I'd argue that because these are searches, [] is actually a valid response, not an error and this should be presented to the user.